tests/gateway/test_client.py

```diff
@@ -181,12 +193,118 @@ def test_list_all_configured_routes(gateway):
     assert routes[0].dict() == {
-        "model": {"name": "gpt-4", "provider": "openai"},
-        "name": "completions-gpt4",
+        "model": {"name": "text-davinci-003", "provider": "openai"},
+        "name": "completions",
         "type": "llm/v1/completions",
     }
     assert routes[1].dict() == {
-        "model": {"name": "gpt-4", "provider": "openai"},
-        "name": "chat-gpt4",
+        "model": {"name": "gpt-3.5-turbo", "provider": "openai"},
+        "name": "chat",
         "type": "llm/v1/chat",
     }
+
+
+def test_client_query_chat(gateway):
+    gateway_client = MlflowGatewayClient(gateway_uri=gateway.url)
+
+    routes = gateway_client.search_routes()
+
+    data = {"messages": [{"role": "user", "content": "How hot is the core of the sun?"}]}
+
+    expected_output = {
+        "candidates": [
+            {
+                "message": {
+                    "role": "assistant",
+                    "content": "The core of the sun is estimated to have a temperature of about "
+                    "15 million degrees Celsius (27 million degrees Fahrenheit).",
+                },
+                "metadata": {"finish_reason": "stop"},
+            }
+        ],
+        "metadata": {
+            "input_tokens": 17,
+            "output_tokens": 24,
+            "total_tokens": 41,
+            "model": "gpt-3.5-turbo-0301",
+            "route_type": "llm/v1/chat",
+        },
+    }
+    mock_response = mock.Mock()
+    mock_response.json.return_value = expected_output
+
+    with mock.patch.object(gateway_client, "_call_endpoint", return_value=mock_response):
+        response = gateway_client.query(route=routes[1].name, data=data)
+
+        assert response == expected_output
+
+
+def test_client_query_completions(gateway):
+    gateway_client = MlflowGatewayClient(gateway_uri=gateway.url)
+
+    routes = gateway_client.search_routes()
+
+    expected_output = {
+        "candidates": [
+            {
+                "text": " car\n\nDriving fast can be dangerous and is not recommended. It is",
+                "metadata": {"finish_reason": "length"},
+            }
+        ],
+        "metadata": {
+            "input_tokens": 7,
+            "output_tokens": 16,
+            "total_tokens": 23,
+            "model": "text-davinci-003",
+            "route_type": "llm/v1/completions",
+        },
+    }
+
+    data = {"prompt": "I like to drive fast in my"}
+
+    mock_response = mock.Mock()
+    mock_response.json.return_value = expected_output
+
+    with mock.patch.object(gateway_client, "_call_endpoint", return_value=mock_response):
+        response = gateway_client.query(route=routes[0].name, data=data)
+        assert response == expected_output
+
+
+def test_client_query_embeddings(gateway):
+    gateway_client = MlflowGatewayClient(gateway_uri=gateway.url)
+
+    routes = gateway_client.search_routes()
+
+    expected_output = {
+        "object": "list",
```

Comment:
What is `object`?

Reply:
I snagged that response structure from your client test suite for openai. I didn't want to try to mock up an actual response from the openai service for an embedding due to the size.

Reply:
I didn't notice this when I copied the example response from OpenAI. Let's keep it.

Reply:
.... and that's the raw response from openai. Fixing!